op-program/client/interop/consolidate.go

```diff
@@ -42,9 +42,37 @@ func ReceiptsToExecutingMessages(depset depset.ChainIndexFromID, receipts ethtyp
 	return execMsgs, curr, nil
 }
 
+type execMessageCacheKey struct {
+	blockHash common.Hash
+	chainID   eth.ChainID
+}
```

Comment:
Strictly speaking, blockHash is all we need here since that commits to the entire chain history, including all the log messages on that chain.  So the same blockHash must always have the same executing messages even if they're on different chains (or really it's just not possible for different chains to have the same block hash).

Reply:
The chain id inclusion is an extra guard against a very contrived case where a malicious sequencer attempts to poison the cache with a dup blockhash. Which might (not really) happen in a roundabout way when both chains undergo block replacement.

Reply:
hmmm, it feels like a real problem if that can happen. The optimistic block hashes are all validated by prior steps in the dispute game and then the replacement block is generated as part of this process, so everything should be trusted inputs. You could definitely poison things if you agree to an invalid prestate but then you're hosed already.

I could definitely be missing something but it makes me nervous to think about having an unvalidated block hash that we'd ever load data from as we can't guarantee we'd have the preimage and chain history for it.

Reply:
Nah, you're right about the same preimage being loaded. Thinking about it again; that is still the case for the contrived block replacement case. We don't need the chain id.